Fixes from PVS Studio static analyzer

dimhotepus opened the pull request:
Some fixes from PVS Studio static analyzer.

Product page: https://www.viva64.com/en/pvs-studio/
How to use for free: https://www.viva64.com/en/b/0600/

dimhotepus committed:
r2: fix c2c1_B computation

t2[1] - t2[1] is (near) zero.

dimhotepus committed:
general: fix memset - zero full buffer

dimhotepus committed:
botlib: Fix large hex letter comparison

dimhotepus committed:
general: initialize used later pointer to default

dimhotepus committed:
qcommon: signed var right shift is undefined behavior

dimhotepus committed:
renderer: add missed else condition

dimhotepus committed:
qcommon: do not read out of buffer

dimhotepus committed:
cgame: do not read out of buffer

dimhotepus committed:
client: volumes are ints, so compare with 0

dimhotepus committed:
server: improve looping semantics

dimhotepus committed:
game: remove already set condition

dimhotepus committed:
renderer2: remove zero side effects code

dimhotepus committed:
general: initialize potentially uninitialized vars

dimhotepus committed:
ui: remove unreachable code

dimhotepus committed:
tools: fix logic condition, skip comments

dimhotepus committed:
cgame: use local loop counter instead of external one

rmarquis commented:
Thanks!

Do you mind changing the categories of the commits, since `misc` is pretty much for the misc stuff only? See [wiki](https://github.com/etlegacy/etlegacy/wiki/How-to-commit-Your-Code#commit-category), f.e. use `cgame`, `ui`, `client`, ... and `general` if the changes affect multiple components.

dimhotepus commented:
> Thanks!
> 
> Do you mind changing the categories of the commits, since `misc` is pretty much for the misc stuff only? See [wiki](https://github.com/etlegacy/etlegacy/wiki/How-to-commit-Your-Code#commit-category), f.e. use `cgame`, `ui`, `client`, ... and `general` if the changes affect multiple components.

I was thinking about misc as `code clean ups` from wiki. Should it be changed to exact touched subsystems?

rmarquis commented:
Yes please.

dimhotepus commented:
In progress, will publish soon.

dimhotepus commented on src/renderer2/tr_shader.c:
```diff
@@ -1537,11 +1537,6 @@ qboolean LoadMap(shaderStage_t *stage, char *buffer)
-	//skip normalmap on sky .... for now
-	else if (!Q_stricmp(token, "sky") )
-	{
-		imageBits |= !IF_NORMALMAP;
```
`!IF_NORMALMAP` is `0`

dimhotepus committed:
qcommon: fix SSE DotProduct

dimhotepus committed:
cgame: eject missed small rock debris

random() % 5 is in [0, 4] range. Allow small rock debris (models/mapobjects/debris/rubble3.md3) for value 5 to be ejected.

dimhotepus committed:
cgame: fix multiline comments parsing

First we check if (fBuffer[p] == '*'), and then, if successfull, check if fBuffer[p] != '*' && fBuffer[p + 1] != '/'), which is already false.

Skip first * and check for */ after that to parse multiline comments correctly.

dimhotepus commented:
@rmarquis Could you, please, review :) ?

dimhotepus committed:
Merge branch 'master' into pvs_studio_fixes

rmarquis commented:
Sorry for the delay. I'm slowly getting through the review. It's mostly great, though there are a few commits I'm not too sure about (as I'm not familiar with some parts of the code).

In particular, the Memset changes look okay to me, but I'll do some more local testing to ensure nothing actually breaks. I'm also not familiar with SSE optimization, but that code isn't actually enabled (see #1442)

dimhotepus commented:
Feel free to remove bad commits.

As for SSE support, it was a bug in `DotProduct`, no value was returned as a result of computation.

rmarquis commented:
Rebased/merged, with very minor modification in some commit messages. Thanks a lot! And sorry it took so long.

dimhotepus commented:
@rmarquis No problem, thank you for review and fixes!
